core/Pos/Binary/Class.hs

```diff
@@ -218,7 +218,9 @@ getTinyVarInt' = do
     if testBit a 7
         then do
             b <- getWord8
-            if | testBit b 7 -> fail "getTinyVarInt': more than 2 bytes"
+            emptyBuf <- isEmpty
```

Comment:
`isEmpty` looks at the whole remaining buffer, so it will return `False` most of the time and mess the decoding up. I'll make a commit with the correct behavior, a sec.